apps/betterangels-backend/shelters/admin.py

```diff
@@ -131,7 +148,192 @@ def _clean_choices(self, field_name: str, model_class: Type[T]) -> list[T]:
+            "storage",
+        ]  # in this case, the many to many fields
+        rowInDict = {}
+        for column in row.keys():
+            rowInDict[column] = row.get(column)
+        if org_name:
+            org, created = Organization.objects.get_or_create(name=org_name)
+            # This process SPA name considering the ManyToMany nature of the field
+            # Gets existing object or makes it if one doesn't exist
+            if spa_name:
+                spa_names = [v.strip() for v in spa_name.split(",")]
+                spa_choices = {i for i in range(1, len(SPAChoices.choices) + 1)}
+                for spa_name in spa_names:
+                    try:
+                        if int(spa_name) in spa_choices:
+                            sp, createdSpa = SPA.objects.get_or_create(name=spa_name)
+                        else:
+                            raise ValueError
+                    except ValueError:
+                        print(f"Row {self.count}: Bad SPA value")
+                        if skip_row_not_val_error:
+                            row["spa"] = None
+                            row["jumpthis"] = True
+                        else:
+                            raise ValidationError(
+                                f"Value in row {self.count} column spa must have a value between 1 and 8"
+                            )
+            # Same idea as the handling for SPA, but uses existing get_or_create_address method in Location class to handle Address creation
+            if addy_name:
+                addy_data = requests.get(
+                    f"https://maps.googleapis.com/maps/api/geocode/json?address={quote(addy_name)}&key={quote(settings.GOOGLE_MAPS_API_KEY)}"
+                )
+                try:
+                    addy_formatted_data = addy_data.json()["results"][0]["formatted_address"]
+                    addy_for_location_method = addy_data.json()["results"][0]
+                    addy_for_location_method["address_components"] = json.dumps(
+                        addy_for_location_method["address_components"]
+                    )
+                    addy_address = Location.get_or_create_address(addy_for_location_method)
+                    if None in [addy_address.street, addy_address.city, addy_address.state, addy_address.zip_code]:
+                        raise IndexError
+                    row["address"] = addy_formatted_data
+                except IndexError:
+                    print(f"Address at {self.count} bad")
+                    if skip_row_not_val_error:
+                        row["address"] = None
+                        row["jumpthis"] = True
+                    else:
+                        raise ValidationError(f"Invalid Address at {self.count}")
+            # This is to process the ManyToMany fields, grabs the data within the CSV and matches it to the proper choice for that column
+            for column in customFields:
```

Comment:
Would it be possible to break out some of this code into smaller functions? We can sync up if you want.